### PR TITLE
Add a minimum wait time check for on_check_publish_block callbacks to…

### PR DESCRIPTION
**Pull request by cintel**
… the BlockPublisher

Signed-off-by: Cian Montgomery <cian.montgomery@intel.com>

### CODE REVIEW

File: validator/sawtooth_validator/journal/journal.py [hunk 1]
```diff
@@ -70,22 +70,34 @@ def stop(self):
             self._exit = True
 
     class _PublisherThread(Thread):
+        CHECK_PUBLISH_BLOCK_FREQUENCY = 0.1
```
Comment (jsmitchell) [inline validator/sawtooth_validator/journal/journal.py]: recommend passing this in as a keyword argument to __init__ with a default. Pass it in from the top level validator class down to whatever instantiates _PublisherThread so that we can establish it via a config file in the future.

**Comment by jsmitchell** [comment]
Can you summarize what problem this is addressing?

File: validator/sawtooth_validator/journal/journal.py (hunk 1 above)
Comment (dcmiddle) [inline validator/sawtooth_validator/journal/journal.py]: I like configurable parameters but I dislike kwargs.

Comment (jsmitchell) [inline validator/sawtooth_validator/journal/journal.py]: To clarify, I'm recommending this `def __init__(self, block_publisher, batch_queue, check_publish_block_frequency=0.1)`

**Comment by cintel** [comment]
This makes the BlockPublisher behavior on checking if a block should be published consistent with respect to incoming batch load from the network.
